core/state_processor.go

```diff
@@ -232,3 +249,114 @@ func ApplyTransaction(config *params.ChainConfig, bc ChainContext, author *commo
+
+	vktKey := tutils.GetTreeKeyVersionWithEvaluatedAddress(addrPoint)
+	leafNodeData := kvm.getOrInitLeafNodeData(vktKey)
+
+	var version [verkle.LeafValueSize]byte
+	leafNodeData.Values[tutils.VersionLeafKey] = version[:]
+
+	var balance [verkle.LeafValueSize]byte
+	for i, b := range acc.Balance.Bytes() {
+		balance[len(acc.Balance.Bytes())-1-i] = b
+	}
+	leafNodeData.Values[tutils.BalanceLeafKey] = balance[:]
+
+	var nonce [verkle.LeafValueSize]byte
+	binary.LittleEndian.PutUint64(nonce[:8], acc.Nonce)
+	leafNodeData.Values[tutils.NonceLeafKey] = balance[:]
+
+	leafNodeData.Values[tutils.CodeKeccakLeafKey] = acc.CodeHash[:]
+
+	// Code size is ignored here. If this isn't an EOA, the tree-walk will call
+	// addAccountCode with this information.
+}
+
+func (kvm *keyValueMigrator) addAccountCode(addr common.Address, codeSize uint64, chunks []byte) {
+	addrPoint := kvm.getAddrPoint(addr)
+
+	vktKey := tutils.GetTreeKeyVersionWithEvaluatedAddress(addrPoint)
+	leafNodeData := kvm.getOrInitLeafNodeData(vktKey)
+
+	// Save the code size.
+	var codeSizeBytes [verkle.LeafValueSize]byte
+	binary.LittleEndian.PutUint64(codeSizeBytes[:8], codeSize)
+	leafNodeData.Values[tutils.CodeSizeLeafKey] = codeSizeBytes[:]
+
+	// The first 128 chunks are stored in the account header leaf.
+	for i := 0; i < 128 && i < len(chunks)/32; i++ {
+		leafNodeData.Values[byte(128+i)] = chunks[32*i : 32*(i+1)]
+	}
+
+	// Potential further chunks, have their own leaf nodes.
+	for i := 128; i < len(chunks)/32; {
+		vktKey := tutils.GetTreeKeyCodeChunkWithEvaluatedAddress(addrPoint, uint256.NewInt(uint64(i)))
+		leafNodeData := kvm.getOrInitLeafNodeData(vktKey)
+
+		j := i
+		for ; (j-i) < 256 && j < len(chunks)/32; j++ {
+			leafNodeData.Values[byte((j-128)%256)] = chunks[32*j : 32*(j+1)]
+		}
+		i = j
+	}
+}
+
+func (kvm *keyValueMigrator) getAddrPoint(addr common.Address) *verkle.Point {
+	if addr == kvm.currAddr {
+		return kvm.currAddrPoint
+	}
+	kvm.currAddr = addr
+	kvm.currAddrPoint = tutils.EvaluateAddressPoint(addr.Bytes())
+	return kvm.currAddrPoint
+}
+
+func (kvm *keyValueMigrator) getOrInitLeafNodeData(stem []byte) *verkle.BatchNewLeafNodeData {
+	stemStr := string(stem)
+	if _, ok := kvm.vktLeafData[stemStr]; !ok {
+		kvm.vktLeafData[stemStr] = &verkle.BatchNewLeafNodeData{
+			Stem:   stem,
+			Values: make(map[byte][]byte),
+		}
+	}
+	return kvm.vktLeafData[stemStr]
+}
+
+func (kvm *keyValueMigrator) migrateCollectedKeyValues(tree *trie.VerkleTrie) error {
+	// Transform the map into a slice.
+	nodeValues := make([]verkle.BatchNewLeafNodeData, 0, len(kvm.vktLeafData))
+	for _, vld := range kvm.vktLeafData {
+		nodeValues = append(nodeValues, *vld)
+	}
+
+	// Create all leaves in batch mode so we can optimize cryptography operations.
+	newLeaves := verkle.BatchNewLeafNode(nodeValues)
+
+	// Insert into the tree.
+	if err := tree.InsertMigratedLeaves(newLeaves); err != nil {
+		return fmt.Errorf("failed to insert migrated leaves: %w", err)
+	}
+
+	return nil
+}
```

Comment:
This is what we called in L173. Basically, we use the new APIs to efficiently create leaves and insert into the tree.